org.lflang/src/org/lflang/generator/CppGenerator.kt

```diff
@@ -0,0 +1,263 @@
+                |find_package(reactor-cpp PATHS ${targetConfig.externalRuntimePath}")
+            """.trimIndent()
+            else """
+                |if(NOT REACTOR_CPP_BUILD_DIR)
+                |    set(REACTOR_CPP_BUILD_DIR "" CACHE STRING "Choose the directory to build reactor-cpp in." FORCE)
+                |endif()
+                |
+                |ExternalProject_Add(dep-reactor-cpp
+                |   PREFIX "$S{REACTOR_CPP_BUILD_DIR}"
+                |   GIT_REPOSITORY "https://github.com/tud-ccc/reactor-cpp.git"
+                |   GIT_TAG "$runtimeVersion"
+                |   CMAKE_ARGS
+                |   -DCMAKE_BUILD_TYPE:STRING=$S{CMAKE_BUILD_TYPE}
+                |   -DCMAKE_INSTALL_PREFIX:PATH=$S{CMAKE_INSTALL_PREFIX}
+                |   -DCMAKE_INSTALL_BINDIR:PATH=$S{CMAKE_INSTALL_BINDIR}
+                |   -DCMAKE_CXX_COMPILER=$S{CMAKE_CXX_COMPILER}
+                |   -DREACTOR_CPP_VALIDATE=${if (targetConfig.noRuntimeValidation) "OFF" else "ON"}
+                |   -DREACTOR_CPP_TRACE=${if (targetConfig.tracing != null) "ON" else "OFF"} 
+                |   -DREACTOR_CPP_LOG_LEVEL=${targetConfig.logLevel.severity}
+                |)
+                |
+                |set(REACTOR_CPP_LIB_DIR "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_LIBDIR}")
+                |set(REACTOR_CPP_BIN_DIR "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_BINDIR}")
+                |set(REACTOR_CPP_LIB_NAME "$S{CMAKE_SHARED_LIBRARY_PREFIX}reactor-cpp$S{CMAKE_SHARED_LIBRARY_SUFFIX}")
+                |set(REACTOR_CPP_IMPLIB_NAME "$S{CMAKE_STATIC_LIBRARY_PREFIX}reactor-cpp$S{CMAKE_STATIC_LIBRARY_SUFFIX}")
+                |
+                |add_library(reactor-cpp SHARED IMPORTED)
+                |add_dependencies(reactor-cpp dep-reactor-cpp)
+                |if(WIN32)
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_IMPLIB "$S{REACTOR_CPP_LIB_DIR}/$S{REACTOR_CPP_IMPLIB_NAME}")
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_LOCATION "$S{REACTOR_CPP_BIN_DIR}/$S{REACTOR_CPP_LIB_NAME}")
+                |else()
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_LOCATION "$S{REACTOR_CPP_LIB_DIR}/$S{REACTOR_CPP_LIB_NAME}")
+                |endif()
+                |
+                |if (APPLE)
+                |   file(RELATIVE_PATH REL_LIB_PATH "$S{REACTOR_CPP_BIN_DIR}" "$S{REACTOR_CPP_LIB_DIR}")
+                |   set(CMAKE_INSTALL_RPATH "@executable_path/$S{REL_LIB_PATH}")
+                |else ()
+                |   set(CMAKE_INSTALL_RPATH "$S{REACTOR_CPP_LIB_DIR}")
+                |endif ()
+            """.trimIndent()
+        }
+            |
+            |set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+            |
+            |set(LF_MAIN_TARGET ${topLevelName})
+            |
+            |add_executable($S{LF_MAIN_TARGET}
+            |    main.cc
+        ${" |    "..sourceFiles.joinToString("\n")}
```

Comment:
This looks nice but I'm not sure how to pronounce it... Reading `..` as `rangeTo` sounds weird here. Maybe using an infix function would be more readable? Something like:
```kt
" |    " indenting sourceFiles.joinToString("\n")
```

Reply:
Yes, I first used an infix function, but I could find a concise name, `indent` just seems too long... I really like the look of the `..` operator here, but yes, I am completely abusing it.

I guess this is something we should discuss with the others and find a solution we all agree on.